### ISSUE TITLE
feat[train]: Create training data loader

### ISSUE THREAD

**Issue by Hamza-Bin-Aamir** [Hamza-Bin-Aamir/PIVOT#33]
Multi-worker data loading (8-16 workers)
Batch size: 2 with gradient accumulation
Pin memory for GPU transfer
Prefetch factor optimization
